src/System.Data.Odbc/src/System/Data/Odbc/OdbcStatementHandle.cs

```diff
@@ -294,29 +294,36 @@ internal ODBC32.RetCode SpecialColumns(string quotedTable)
                                            short unique,
                                            short accuracy)
         {
-            ODBC32.RetCode retcode = Interop.Odbc.SQLStatisticsW(this,
-                                                                        tableCatalog,
-                                                                        ODBC.ShortStringLength(tableCatalog),
-                                                                        tableSchema,
-                                                                        ODBC.ShortStringLength(tableSchema),
-                                                                        tableName,
-                                                                        ODBC.ShortStringLength(tableName),
-                                                                        unique,
-                                                                        accuracy);
+            ODBC32.RetCode retcode;
+
+            // MDAC Bug 75928 - SQLStatisticsW damages the string passed in
+            // To protect the tablename we need to pass in a copy of that string
+
+            IntPtr pwszTableName = Marshal.StringToCoTaskMemUni(tableName);
+            try
+            {
+                retcode = Interop.Odbc.SQLStatisticsW(this,
+                                                      tableCatalog,
+                                                      ODBC.ShortStringLength(tableCatalog),
+                                                      tableSchema,
+                                                      ODBC.ShortStringLength(tableSchema),
+                                                      pwszTableName,
+                                                      ODBC.ShortStringLength(tableName),
+                                                      unique,
+                                                      accuracy);
+            }
+            finally
+            {
+                Marshal.FreeCoTaskMem(pwszTableName);
+            }
 
             ODBC.TraceODBC(3, "SQLStatisticsW", retcode);
             return retcode;
         }
 
         internal ODBC32.RetCode Statistics(string tableName)
         {
-            ODBC32.RetCode retcode = Interop.Odbc.SQLStatisticsW(this,
-            null, 0, null, 0,
-            tableName, ODBC.ShortStringLength(tableName),
-            (short)ODBC32.SQL_INDEX.UNIQUE,
-            (short)ODBC32.SQL_STATISTICS_RESERVED.ENSURE);
-            ODBC.TraceODBC(3, "SQLStatisticsW", retcode);
```

Comment:
The tracing isn't needed?

Reply:
It's still there in the fuller overload (which this overload calls into).